File: LC_PYTHON/Move_all_negitive_ele_end.py
#User function Template for python3

class Solution:
    def segregateElements(self, arr):
        # Your code goes here
        temp = []
        for i in arr:
            if(i>=0):
                temp.append(i)

        for i in arr:
            if(i<0):
                temp.append(i)

        for i in range(len(temp)):
            arr[i] = temp[i]
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3


def main():

    T = int(input())

    while (T > 0):
        a = [int(x) for x in input().strip().split()]
        ob = Solution()
        ob.segregateElements(a)
        print(*a)

        T -= 1


if __name__ == "__main__":
    main()

# } Driver Code Ends

File: LC_JAVA/Move_all_negitive_ele_end.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.lang.*;
import java.util.*;

public class Move_all_negitive_ele_end {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = Integer.parseInt(sc.nextLine());

        while (T > 0) {
            String[] input = sc.nextLine().trim().split(" ");
            int[] a = new int[input.length];
            for (int i = 0; i < input.length; i++) {
                a[i] = Integer.parseInt(input[i]);
            }
            Solution ob = new Solution();
            ob.segregateElements(a);
            for (int num : a) {
                System.out.print(num + " ");
            }
            System.out.println();
            T--;
        }
        sc.close();
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public void segregateElements(int[] arr) {
        // Your code goes here
       int n = arr.length;
       int[] temp = new int[n];
       int index = 0;
        for(int i=0;i<arr.length;i++){
            if(arr[i]>=0){
                temp[index++]=arr[i];
            }
        }

        for(int i=0;i<arr.length;i++){
            if(arr[i]<0){
                temp[index++]=arr[i];
            }
        }

        for(int i=0;i<temp.length;i++){
            arr[i] = temp[i];
        }
    }
}
